lib/rucio/core/request.py

```diff
@@ -1390,6 +1396,20 @@ def observe(
                 if state == RequestState.DONE:
                     record.files_done += 1
                     record.bytes_done += file_size
+
+                    if submitted_at is not None and started_at is not None:
+                        wait_time = (started_at - submitted_at).total_seconds()
+
+                        # avg_wait_time averages across a tumbling window, so reset avg every few hours
+                        hours_window = 6
+                        if self.avg_wait_time_start_timestamp < (now - datetime.timedelta(hours=hours_window)):
+                            self.avg_wait_time = 0
+                            self.transfers_done = 0
+                            self.avg_wait_time_start_timestamp = now
+
+                        self.avg_wait_time = ((self.avg_wait_time * self.transfers_done) + wait_time) / (self.transfers_done + 1)
+                        self.transfers_done += 1
+                        METRICS.gauge('avg_wait_time').set(self.avg_wait_time)
```

Comment:
We do not add our gauge metric with any filters (such as by src id or activity), since this gauge will be a global value corresponding to the overall average wait time for an entire topology. We imagine that one prometheus, this gauge will show up as a line graph or time plot, where the x-axis is the timestamp when the gauge metric was pulled, and y-axis is the avg_wait_time.